security/medusa/l4-constable/med_cache.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Medusa L4 memory cache interface
+ *
+ * (C) 2018, 2020 Roderik Ploszek <roderik.ploszek@gmail.com>
+ */
+
+#include <linux/slab.h>
+
+#define FAT_PTR_OFFSET_TYPE uint32_t
+#define FAT_PTR_OFFSET sizeof(FAT_PTR_OFFSET_TYPE)
+
+// Array for memory caches
+int cache_array_size = 0;
+static struct kmem_cache **med_cache_array;
+
+/**
+ * Computes binary logarithm rounded up
+ */
+static inline int get_log_index(int in) {
```

Comment:
I thought how we could simplify this one. I see that every time we need to pass into function ``SIZE + FAT_PTR_SIZE``. I believe the logic is to compute cache_index so my idea would be to rename: ``get_log_index`` -> ``_log2_rounded_up``. Then create new function ``get_mem_cache_index(size_t)`` which will call ``_log2_rounded_up`` and also will add automatically FAT_PTR_SIZE to argument. Then replace all references of ``get_log_index`` in the allocation functions for ``get_mem_cache_index``. My intention is to not bother people with the implementation detail like how the index is computed -> that it is actually a binary log, but rather to know what it represents -> index for memory cache.

In addition if you want to make it unit testable, you can create a lower level function ``_get_mem_cache_index(size_t size, size_t offset)``, but I will leave it up to you.

What do you think?